src/framework/components/element/markup.js

```diff
@@ -0,0 +1,417 @@
+                                // the slash without interpretation and continue
+                                this._output("\\");
+                                break;
+                        }
+                        break;
+                    default:
+                        this._store();
+                        break;
+                }
+            }
+        },
+
+        // read tag block
+        _tag: function () {
+            while (true) {
+                switch (this._cur) {
+                    case null:
+                        this._error = "unexpected end of input reading tag";
+                        return this.ERROR_TOKEN;
+                    case "[":
+                        this._store();
+                        return this.OPEN_BRACKET_TOKEN;
+                    case "]":
+                        this._store();
+                        this._mode = "text";
+                        return this.CLOSE_BRACKET_TOKEN;
+                    case "=":
+                        this._store();
+                        return this.EQUALS_TOKEN;
+                    case " ":
+                    case "\t":
+                    case "\n":
+                    case "\r":
+                    case "\v":
+                    case "\f":
+                        return this._whitespace();
+                    case "\"":
+                        return this._string();
+                    default:
+                        if (!this._isIdentifierSymbol(this._cur)) {
+                            this._error = "unrecognized character";
+                            return this.ERROR_TOKEN;
+                        }
+                        return this._identifier();
+                }
+            }
+        },
+
+        _whitespace: function () {
+            this._store();
+            while (" \t\n\r\v\f".indexOf(this._cur) !== -1) {
+                this._store();
+            }
+            return this.WHITESPACE_TOKEN;
+        },
+
+        _string: function () {
+            this._next();       // skip "
+            while (true) {
+                switch (this._cur) {
+                    case null:
+                        this._error = "unexpected end of input reading string";
+                        return this.ERROR_TOKEN;
+                    case "\"":
+                        this._next();           // skip "
+                        return this.STRING_TOKEN;
+                    default:
+                        this._store();
+                        break;
+                }
+            }
+        },
+
+        _identifier: function () {
+            this._store();
+            while (this._cur !== null &&
+                    this._isIdentifierSymbol(this._cur)) {
+                this._store();
+            }
+            return this.IDENTIFIER_TOKEN;
+        },
+
+        _isIdentifierSymbol: function (s) {
+            return s.length === 1 && (s.match(/[A-Z|a-z|0-9|_|-|/]/) !== null);
```

Comment:
Consider making this regex a constant to avoid recreating it every time

Reply:
done